File: sdk/typescript/examples/run_stream.ts
import { AegisClient, DoneEvent, ErrorEvent, ProofEvent, StdoutEvent } from "@aegis/sdk";

const client = new AegisClient();
for await (const event of client.stream({ language: "bash", code: "echo streamed" })) {
  if (event instanceof StdoutEvent) {
    process.stdout.write(event.chunk);
  } else if (event instanceof ProofEvent) {
    console.log(`receiptPath=${event.proofBundle.receiptPath}`);
  } else if (event instanceof DoneEvent) {
    console.log(`done exit=${event.exitCode} durationMs=${event.durationMs}`);
  } else if (event instanceof ErrorEvent) {
    console.log(`stream error: ${event.error}`);
  }
}

File: sdk/python/examples/run_stream.py
from aegis import AegisClient, DoneEvent, ErrorEvent, ProofEvent, StdoutEvent

client = AegisClient()
for event in client.stream(language="bash", code="echo streamed"):
    if isinstance(event, StdoutEvent):
        print(event.chunk, end="")
    elif isinstance(event, ProofEvent):
        print(f"receipt_path={event.proof_bundle.receipt_path}")
    elif isinstance(event, DoneEvent):
        print(f"done exit={event.exit_code} duration_ms={event.duration_ms}")
    elif isinstance(event, ErrorEvent):
        print(f"stream error: {event.error}")
